apps/explorer/src/lib/ui/utils/generateValidatorsTableColumns.tsx

```diff
@@ -257,20 +259,81 @@ export function generateValidatorsTableColumns({
                         </TableCellBase>
                     );
                 }
+
                 return (
                     <TableCellBase>
                         <Badge
                             type={
-                                atRisk === null && isCommitteeMember
-                                    ? BadgeType.PrimarySoft
-                                    : BadgeType.Neutral
+                                atRisk === null
+                                    ? BadgeType.Success
+                                    : atRisk > 1
+                                      ? BadgeType.Warning
+                                      : BadgeType.Error
                             }
                             label={label}
                         />
                     </TableCellBase>
                 );
             },
         },
+        {
+            header: 'Current Epoch Rewards',
+            accessorKey: 'isEarningCurrent',
+            id: 'isEarningCurrent',
+            enableSorting: true,
+            cell({ row }) {
+                const isCommitteeMember = committeeMembers.find(
+                    (committeeMemberAddress) => committeeMemberAddress === row.original.iotaAddress,
+                );
+                const label = isCommitteeMember ? 'Earning' : 'Not Earning';
+                return (
+                    <TableCellBase>
+                        <Badge
+                            type={isCommitteeMember ? BadgeType.PrimarySoft : BadgeType.Neutral}
+                            label={label}
+                        />
+                    </TableCellBase>
+                );
+            },
+        },
+        {
+            header: 'Next Epoch Rewards',
+            accessorKey: 'isEarningNext',
+            id: 'isEarningNext',
+            enableSorting: true,
+            cell({ row }) {
+                const { atRisk, isPending, isCommitteeMember } = determineRisk(
+                    committeeMembers,
+                    atRiskValidators,
+                    row,
+                );
+
+                const sortedActiveValidators = activeValidators.toSorted(sortByStakingBalanceDesc);
+                const topValidators = sortedActiveValidators.slice(0, maxCommitteeMembers);
```

Comment:
Can you move this out of the cell rendering? I feel like this is gonna be an absolute performance killer otherwise

Reply:
[c50ad65](https://github.com/iotaledger/iota/pull/6601/commits/c50ad657c500c42e8d33079dbbce66d5f9300012)